api/internal/utils/timedcall_test.go

```diff
@@ -62,3 +63,15 @@ func TestTimedCallSlowWithError(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func TestTimedCallGoroutineLeak(t *testing.T) {
+	beforeGroutine := runtime.NumGoroutine()
+	_ = TimedCall("expect no goroutine leaks", time.Second, func() error {
+		time.Sleep(2 * time.Second)
+		return fmt.Errorf("function done")
+	})
+	afterGoroutine := runtime.NumGoroutine()
+	if beforeGroutine != afterGoroutine {
+		t.Fatal("goroutine leak")
+	}
```

Comment:
Small suggestion:
```suggestion
    require.NotEqual(t, beforeGoroutine, afterGoroutine, "goroutine leak")
```

Reply:
Hi @stormqueen1990 , I think it should be `require.Equal` here.  Because we modified the `unbuffered channel` to be buffered, it should be equal when the test is done here.

Reply:
Yes! I forgot to invert the test when I typed, sorry about that 😅